certora/specs/ReentrancyView.spec

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2025 Morpho Association
+
+methods {
+    function multicall(bytes[]) external => NONDET DELETE;
+
+    function _.realAssets() external => ignoredUintStaticcall() expect(uint256);
+
+    function _.balanceOf(address) external => ignoredUintStaticcall() expect(uint256);
+
+    function _.canReceiveShares(address) external => ignoredBoolStaticcall() expect(bool);
+    function _.canSendShares(address) external => ignoredBoolStaticcall() expect(bool);
+    function _.canReceiveAssets(address) external => ignoredBoolStaticcall() expect(bool);
+    function _.canSendAssets(address) external => ignoredBoolStaticcall() expect(bool);
+}
+
+function ignoredBoolStaticcall() returns bool {
+    ignoredStaticcall = true;
+    bool value;
+    return value;
+}
+
+function ignoredUintStaticcall() returns uint256 {
+    ignoredStaticcall = true;
+    uint256 value;
+    return value;
+}
+
+persistent ghost bool ignoredStaticcall;
+
+// True when at least one slot was written.
+persistent ghost bool storageChanged;
+
+// True when at least one STATICCALL is executed after a storage change.
+persistent ghost bool staticCallAfterSStore;
+
+// True when at least one slot is changed after a STATICCALL is executed after a storage change.
+persistent ghost bool staticCallUnsafe;
+
+hook ALL_SSTORE(uint _, uint _) {
+    storageChanged = true;
+    if (staticCallAfterSStore) {
+        staticCallUnsafe = true;
+    }
+}
+
+hook STATICCALL(uint256 g, address addr, uint256 argsOffset, uint256 argsLength, uint256 retOffset, uint256 retLength) uint256 rc {
+    if (!ignoredStaticcall && storageChanged && addr != 0x1 && addr != currentContract) {
```

Comment:
Why != 0x1 and why exclude self-static call?

Reply:
0x1 is ecrecover, it's just doing a pure computation so we can safely ignore it. For the self-static call I think it's an artifact of the Reentrancy.spec, but I think we can remove it

Reply:
I added a comment for 0x1, and I removed the exclusion of self-static calls